iowrite.cpp

```diff
@@ -1111,7 +1111,7 @@ bool writeVelocitySpace(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
 	 // Loop over AMR levels
 	 uint startindex=1;
 	 uint endindex=1;
-	 for (int AMR = 0; AMR <= P::amrMaxSpatialRefLevel; AMR++) {
+	 for (int AMR = 0; AMR <= static_cast<int>(P::amrMaxSpatialRefLevel); AMR++) {
```

Comment:
In my opinion, unsigned types should be avoided due to unintuitive behavior. In case of loops I think compilers can also optimize signed integers better since overflow is UB. The best option here would probably be changing the declaration of `P::amrMaxSpatialRefLevel`  instead of casting elsewhere.

In this case the C cast should be identical since we're working with basic types, but from what I've heard `static_cast` should be used basically always due to compile-time checks and whatnot.

Reply:
Indeed, I agree changing ```P::amrMaxSpatialRefLevel``` to int would be the preferable solution there.
Also: I suppose the next line would much more reasonably be written as
```
uint AMRm = 1u<<AMR;
```
instead of typeconverting back and forth between ints and floats.

Reply:
Both done